### ISSUE TITLE
Protected Recovery Phrases

### ISSUE THREAD

**Issue by lastmjs** [dfinity/internet-identity#657]
Hi!

I'll be working on the frontend for protected recovery phrases as discussed in multiple places:
* https://forum.dfinity.org/t/immediate-action-to-protect-internet-identity-w-seed-phrases/12528
* https://forum.dfinity.org/t/protected-recovery-phrases-plan-and-progress/12950

As Internet Identity is now accepting [outside contributions](https://forum.dfinity.org/t/internet-identity-now-accepts-external-contributions-on-github/13227), this will be a great learning experience for us all. I've already been in contact with DFINITY and been approved to work on this.

I'll be generally following the suggestions found here:
* https://forum.dfinity.org/t/immediate-action-to-protect-internet-identity-w-seed-phrases/12528/19?u=lastmjs
* https://forum.dfinity.org/t/immediate-action-to-protect-internet-identity-w-seed-phrases/12528/61?u=lastmjs

Here's the general plan:
1. Add a new lock icon next to the recovery phrase under Anchor Management -> Recovery Mechanisms
2. The lock icon will be unlocked for all existing recovery phrases, and then will be either locked/unlocked depending on user locking/unlocking moving forward
3. If the user clicks an unlocked lock icon
    1. The user will be shown a dialog explaining what locking their recovery phrase means
    2. The user will have to enter their recovery phrase to perform the locking action
    3. The user will be asked to confirm their choice to lock their recovery phrase
    4. Once confirmed, an update call will be made to lock the recovery phrase, but it will not succeed unless the call is made with a signature from the recovery phrase keypair
4. If the user clicks a locked icon
    1. The user will be shown a dialog explaining what unlocking their recovery phrase means
    2. The user will have to enter their recovery phrase to perform the unlocking action
    3. The user will be asked to confirm their choice to unlock their recovery phrase
    4. Once confirmed, an update call will be made to unlock the recovery phrase, but it will not succeed unless the call is made with a signature from the recovery phrase keypair
5. If the user clicks the remove recovery phrase icon
    1. ?? Should we ask the user to enter their recovery phrase on the frontend no matter what ??
    2. If the recovery phrase is locked then an update call will be made to remove the recovery phrase, but it will not succeed unless the call is made with a signature from the recovery phrase keypair
    3. If the recovery phrase is unlocked then an update call will be made to remove the recovery phrase

**Comment by dostro** [comment]
@lastmjs looks good with a couple questions

Should we also add the checkbox prompting user to protect this recovery phrase [at the point of its presentation](https://forum.dfinity.org/t/immediate-action-to-protect-internet-identity-w-seed-phrases/12528/17?u=dostro)?

```
5. If the user clicks the remove recovery phrase icon
?? Should we ask the user to enter their recovery phrase on the frontend no matter what ??
```
By this, do you mean whether or not it's set to `protected`? It's interesting because it would probably save you some development time and `might` deter attackers, but I would think it'd ultimately be too confusing for users why this dialog is being displayed without any functional reason.

**Comment by lastmjs** [comment]
> Should we also add the checkbox prompting user to protect this recovery phrase [at the point of its presentation](https://forum.dfinity.org/t/immediate-action-to-protect-internet-identity-w-seed-phrases/12528/17?u=dostro)?

Yes

**Comment by lastmjs** [comment]
I could use some guidance from the II team on how to make the lock and unlock icons. It looks like you are using hard-coded SVGs for icons as found here: https://github.com/dfinity/internet-identity/blob/main/src/frontend/src/components/icons.ts

**Comment by nmattia** [comment]
I'll +1 @peterpeterparker 's comment with emphasis on "... mentioning the source as comment in the code". For confirmation on where the icons come from, maybe a word from @nomeata or @kritzcreek who were around when II was first developed?

**Comment by kritzcreek** [comment]
As far as I remember they were made by our own designers (when we still had those).

**Comment by lastmjs** [comment]
@peterpeterparker thanks! We've actually simplified the implementation quite a bit and hopefully won't need new icons for now.

I've run into something interesting and I hope someone knows the answer off the top of their head: Is the anchor number part of the seed phrase?

I was able to remove a recovery phrase by entering in the correct recovery phrase but changing the anchor number to something else, and I wonder if that's a problem

**Comment by plitzenberger** [comment]
@lastmjs I'm relatively sure that the user number is not part of the SeedPhrase "device". Check this for reference:

https://github.com/dfinity/internet-identity/blob/3ac41913b934913220bfa015eebf0a7e2bdfc8a5/src/frontend/src/flows/recovery/inputSeedPhrase.ts#L72-L87

**Comment by nmattia** [comment]
Closed in #721
